cms/db/contest.py

```diff
@@ -193,6 +196,21 @@ class Contest(Base):
         nullable=False,
         default=datetime(2100, 1, 1))
 
+    # Beginning and ending of the contest anaylsis mode.
+    analysis_enabled = Column(
+        Boolean,
+        nullable=False,
+        default=False,
+    )
+    analysis_start = Column(
+        DateTime,
+        nullable=False,
+        default=datetime(2100, 1, 1))
+    analysis_stop = Column(
+        DateTime,
+        nullable=False,
+        default=datetime(2100, 1, 1))
```

Comment:
Why not just "analysis_enabled"? Is there a good reason to have "analysis_start" and "analysis_stop"? I would just have CWS show analysis mode (i.e. the possibility to download submissions and to test new submission but without affecting the score) *if and only if* "analysis_enabled" is true, regardless of the date.